### PR TITLE
move OsStr helpers to a separate file

### PR DESCRIPTION
**Pull request by JOE1994**
Moved OsStr read/write functions from `src/helpers.rs` to `src/shims/os_str.rs`, in order to keep `src/helpers.rs` from bloating too much.

### CODE REVIEW

File: src/shims/os_str.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::ffi::{OsStr, OsString};
+use std::iter;
+#[cfg(unix)]
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+#[cfg(windows)]
+use std::os::windows::ffi::{OsStrExt, OsStringExt};
```
Comment (RalfJung) [inline src/shims/os_str.rs]: Please keep these conditional imports visually separated, like they were in the original file.

Comment (JOE1994) [reply src/shims/os_str.rs]: Should it be something like this? (Assuming alphabetical order of imports)
```rust
use std::iter;

#[cfg(unix)]
use std::os::unix::ffi::{OsStrExt, OsStringExt};
#[cfg(windows)]
use std::os::windows::ffi::{OsStrExt, OsStringExt};

use std::path::{Path, PathBuf};
```

Comment (RalfJung) [reply src/shims/os_str.rs]: No, I'd first group all the regular imports (alphabetically sorted, if you want), and then separately the conditional ones.

```rust
use std::iter;
use std::path::{Path, PathBuf};

#[cfg(unix)]
use std::os::unix::ffi::{OsStrExt, OsStringExt};
#[cfg(windows)]
use std::os::windows::ffi::{OsStrExt, OsStringExt};
```

**Comment by RalfJung** [comment]
Thanks, this looks good. :)
@bors r+

**Pull request merged**